challengeMk2/ViewModels/SystemDetailCarouselViewModel.cs

```diff
@@ -1,185 +1,145 @@
-                BodyCount = currentSystem.BodyCount,
-                Name = detailedSystem.Name,
-                RequirePermit = detailedSystem.RequirePermit,
-                PermitName = detailedSystem.PermitName,
-                Information = detailedSystem.Information,
-                PrimaryStar = detailedSystem.PrimaryStar,
-                Coords = detailedSystem.Coords
-            };
-
-            return systemWithId;
-        }
-
-        void SwitchBanner(int position)
-        {
-            CurrentBanner = banners[position];
-        }
-
-        public async Task InitializeAsync(StarSystem selectedSystem)
-        {
-            SetCurrentSystem(selectedSystem);
-
-            await UpdateSystemData();
-        }
-    }
-}
+using System.Collections.ObjectModel;
+using ChallengeMk2.Models;
+using Xamarin.Forms;
+
+
+namespace ChallengeMk2.ViewModels
+{
+    public class SystemDetailCarouselViewModel : BaseViewModel
+    {
+        readonly string[] banners;
+
+
+        public SystemDetailCarouselViewModel()
+        {
+            Title = "Star System Details";
+
+            SystemInfos = new ObservableCollection<StarSystem>();
+
+            banners = new string[]
+            {
+                "BannerDetail_01",
+                "BannerDetail_08",
+                "BannerDetail_04",
+                "BannerDetail_07"
+            };
+
+            SwitchBannerCommand = new Command<int>(p => SwitchBanner(p));
+            currentBanner = banners[0];
+        }
+
+
+        StarSystem currentSystem;
+        public StarSystem CurrentSystem
+        {
+            get => currentSystem;
+            set
+            {
+                SetProperty<StarSystem>(ref currentSystem, value);
+            }
+        }
+
+        StarSystem detailedSystem;
+        public StarSystem DetailedSystem
+        {
+            get => detailedSystem;
+            set
+            {
+                SetProperty<StarSystem>(ref detailedSystem, value);
+            }
+        }
+
+        int currentBodyCount;
+        public int CurrentBodyCount
+        {
+            get => currentBodyCount;
+            set
+            {
+                SetProperty<int>(ref currentBodyCount, value);
+            }
+        }
+
+        double currentDistance;
+        public double CurrentDistance
+        {
+            get => currentDistance;
+            set
+            {
+                SetProperty<double>(ref currentDistance, value);
+            }
+        }
+
+        string currentBanner;
+        public string CurrentBanner
+        {
+            get => currentBanner;
+            set
+            {
+                SetProperty<string>(ref currentBanner, value);
+            }
+        }
+
+        public Command SwitchBannerCommand { get; set; }
+
+        public ObservableCollection<StarSystem> SystemInfos { get; set; }
+
+
+        void SetCurrentSystem(StarSystem selectedSystem)
+        {
+            currentSystem = selectedSystem;
```

Comment:
If you assign a new value to a private field, this will not trigger SetProperty so changes won't be reflected on your view.

Reply:
You put your finger on it ! A lot of old code here. 
I clean up that mess and simplify the code. Remove unused props.